File: src/lib/api/codexSubagents.ts
import { invoke } from "@tauri-apps/api/core";
import type { CodexSubagentSettingsView } from "@/types";

export const codexSubagentsApi = {
  getSettings(): Promise<CodexSubagentSettingsView> {
    return invoke("get_codex_subagent_settings");
  },

  saveSettings(
    model: string | null,
    reasoningEffort: string | null,
  ): Promise<CodexSubagentSettingsView> {
    return invoke("save_codex_subagent_settings", {
      model,
      reasoningEffort,
    });
  },
};

File: src-tauri/src/commands/codex_subagents.rs
#![allow(non_snake_case)]

use crate::codex_subagents::{self, CodexSubagentSettingsView};

#[tauri::command]
pub async fn get_codex_subagent_settings() -> Result<CodexSubagentSettingsView, String> {
    codex_subagents::get_settings_view().map_err(|error| error.to_string())
}

#[tauri::command]
pub async fn save_codex_subagent_settings(
    model: Option<String>,
    reasoningEffort: Option<String>,
) -> Result<CodexSubagentSettingsView, String> {
    codex_subagents::save_settings(model, reasoningEffort).map_err(|error| error.to_string())
}
